### DISCUSSION TITLE
Amazon debuting replacement to MWS: Selling Partners API

### DISCUSSION THREAD

**Discussion by GriceTurrble** [python-amazon-mws/python-amazon-mws#212] [General]
Announcement can be found here: https://www.programmableweb.com/news/amazon-to-debut-new-selling-partner-api-sp-api/brief/2020/09/10

Docs showing up on GitHub, here: https://github.com/amzn/selling-partner-api-docs

API models and helper client code, here: https://github.com/amzn/selling-partner-api-models

Per announcement:

> Amazon has announced a new API that is intended to serve as a next-generation resource in the advancement of everything previously handled by the Amazon Marketplace Web Service (MWS) APITrack this API. The new Amazon Selling Partner API (SP-API) will provide all the legacy functionality offered to developers via MWS in addition to key improvements.

> The new SP-API is REST-based and outputs in JSON. Additionally, support for OAuth 2.0 via Login with Amazon is included. Amazon notes that one of the most requested features for the MWS API, an endpoint for testing, will be included in the SP-API. Going forward developers that already use the MWS API can continue to do so, but Amazon will be only be updating the API for business-critical changes. To access the latest features developers will want to migrate to the SP-API. 

> Although the new API is not yet generally available, Amazon expects this level of access to be imminent. Make sure to check back as this article will be updated when general availability is announced. For more details, developers can check the documentation for the MWS API.

**TL;DR**: MWS will be legacy code in due time, with SP-API replacing it with a REST API that accepts and responds in JSON.

---

We'll be following this as it develops and offer support for devs who continue to use MWS, as well as watching out for how we can best leverage to newer SP-API going forward.

**Comment by jerch** [comment]
Those are really great news. Overall JSON handling is much more light-weighted, no reason to point heavy XML guns on amazon assets anymore.

Lets hope they stick to the promise above to provide all legacy functionality also with the JSON version. Also oauth will make automated integration much easier. Not sure whether to cry or to laugh about a testing endpoint.

Kinda wonder how they plan to migrate ppl to the new thingy once it is rolled out. Hope they give enough months (years?) to catch up. Well, we will see...

**Reply by Bobspadger** [reply]
I have waited years to hear they are going to finally have a test system - my goodness me, finally!

**Comment by 854350999** [comment]
Amazon Advertising API is already in this mode, and I have also used this new mode, JSON data is more convenient than usual, so I also look forward to integrating the entire background data into one.
